x-pack/plugins/fleet/server/services/preconfiguration.ts

```diff
@@ -143,7 +143,8 @@ export async function ensurePreconfiguredPackagesAndPolicies(
       if (!created) {
         if (!policy?.is_managed) return { created, policy };
         const configTopLevelFields = omit(preconfiguredAgentPolicy, 'package_policies', 'id');
-        const currentTopLevelFields = omit(policy, 'package_policies');
+        const currentTopLevelFields = pick(policy, ...Object.keys(configTopLevelFields));
+
         if (!isEqual(configTopLevelFields, currentTopLevelFields)) {
```

Comment:
Do you think it will make sense to have a non regression test here?  and maybe extract this to a `hasPreconfiguredPolicyChanged()` or better naming?